.tekton/tasks/update-dependencies.yaml

```diff
@@ -36,17 +36,29 @@ spec:
         && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
         && apt update \
         && apt install gh -y
-
+        apt install jq -y
+        
         gh auth login --with-token
 
         git ls-remote --exit-code --heads origin currency-bot >/dev/null 2>&1
-        EXIT_CODE=$?
+        BRANCH_EXISTS=$?
 
-        if [[ $EXIT_CODE != '0' ]]; then          
+        if [[ $BRANCH_EXISTS != 0 ]]; then
           echo "Running patch and minor updates..."
           MAJOR_UPDATES_MODE=false BRANCH=currency-bot node bin/currency/update-currencies.js    
         else
-          echo "Skipping patch/minor updates. Branch exists."
+          echo "currency-bot branch exists. Checking for open pull requests.."
+
+          OPEN_PRS=$(gh pr list --head currency-bot --base main --state open --json number | jq 'length')
+
+          if [[ $OPEN_PRS -eq 0 ]]; then
+            echo "No open PRs found. Deleting branch currency-bot..."
+            git push origin --delete currency-bot
+            echo "Running patch and minor updates..."
+            MAJOR_UPDATES_MODE=false BRANCH=currency-bot node bin/currency/update-currencies.js      
```

Comment:
suggestion: To avoid having to use the same command twice:

in line 45:
```
OPEN_PRS=$(gh pr list --head currency-bot --base main --state open --json number | jq 'length')
git ls-remote --exit-code --heads origin currency-bot >/dev/null 2>&1
BRANCH_EXISTS=$?

# auto cleanup in case we forgot to manually delete the branch
if BRANCH EXISTS AND NO OPEN PULL REQUESTS
  git push origin --delete currency-bot

...continue with normal logic...
```